plugin-gradle/src/test/java/com/diffplug/gradle/spotless/GradleIntegrationTest.java

```diff
@@ -58,9 +58,9 @@ public void gitAttributes() throws IOException {
 
 	protected final GradleRunner gradleRunner() throws IOException {
 		return GradleRunner.create()
-				// Test against Gradle 2.14.1 in order to maintain backwards compatibility.
-				// https://github.com/diffplug/spotless/issues/161
-				.withGradleVersion("2.14.1")
+				// Gradle 4.9 required for Task Configuration Avoidance
+				// (https://github.com/diffplug/spotless/issues/269)
+				.withGradleVersion("4.9")
```

Comment:
I would actually keep this as `withGradleVersion("2.14.1")`.  That's how we make sure that our legacy support doesn't break.

Then, I would make a new `GradleTaskConfigurationAvoidanceTest`, which uses gradle 4.9, and asserts that tasks are actually created lazily.  I don't see a need to introduce the GradleTest plugin, but maybe @eriwen knows about value that I'm missing.

Reply:
@nedtwigg @jbduncan Is there an easy way to run all the tests twice with both versions?

Reply:
There might be, but our gradle tests are already pretty darn slow.  None of our steps rely on Gradle, so it's really just the `SpotlessExtension` logic that we need to test.  We compile against `4.9`, so there's no risk that we'll use API that doesn't exist in 4.9.  But we might accidentally use API that was added after `2.14`, and that's what we'll catch by always running against `2.14` - problems with `4.9` will be caught at compile-time.

For gradle == 4.9, it's a good idea to test and make sure that the task avoidance is doing what we intend it to do.  That will have the added bonus of being an integration test for the `4.9+` logic in general.

Reply:
That all sounds good to me. I will revert this change and create `GradleTaskConfigurationAvoidanceTest` as requested. :)